test/healthcheck/HealthChecker.test.ts

```diff
@@ -1145,4 +1145,36 @@ describe('Health Checker test suite', () => {
     const expected = "{\"status\":\"STOPPING\",\"checks\":[{\"name\":\"checkone\",\"state\":\"STOPPED\",\"data\":{\"reason\":\"\"}},{\"name\":\"checktwo\",\"state\":\"STOPPING\",\"data\":{\"reason\":\"\"}}]}"
     expect(result).to.equal(expected, `Should return: ${expected}, but returned: ${result}`);
   });
+
+  it('Readiness should return DOWN with a failing Readiness check not throwing an error' , async() => {
+    let healthcheck = new HealthChecker();
+    const readinessPromise = () => new Promise<void>((resolve, reject) => {
+      reject("Readiness Failure");
+    });
+
+    let check = new ReadinessCheck("readinessCheck", readinessPromise);
+    healthcheck.registerReadinessCheck(check);
+    let status = await healthcheck.getReadinessStatus();
+    status = await healthcheck.getReadinessStatus();
+    const result = JSON.stringify(status);
```

Comment:
If you did the comparison as JSON-to-JSON that really long line below could be more readable, and any test failure would list the keys that didn't match, rather than just dump long strings to console.